common/define-grammar.js

```diff
@@ -131,6 +131,27 @@ module.exports = function defineGrammar(dialect) {
         optional($._initializer)
       ),
 
+      // override original catch_clause, add optional type annotation
+      catch_clause: $ => seq(
```

Comment:
maybe add the previous parameter here? to better show that it's an overriding rule (or $_previous to indicate we don't use it).